internal/pkg/log/config.go

```diff
@@ -2,45 +2,96 @@
 package log
 
 import (
+	"fmt"
 	"log/slog"
+	"strings"
 	"time"
+
+	"github.com/internetarchive/Zeno/internal/pkg/config"
 )
 
 // Config defines the configuration for the logging package
 type Config struct {
-	FileConfig               *LogfileConfig
-	StdoutEnabled            bool
-	StdoutLevel              slog.Level
-	StderrEnabled            bool
-	StderrLevel              slog.Level
-	RotateLogFile            bool
-	RotatePeriod             time.Duration
-	ElasticsearchConfig      *ElasticsearchConfig
-	RotateElasticSearchIndex bool
+	FileConfig          *LogfileConfig
+	StdoutEnabled       bool
+	StdoutLevel         slog.Level
+	StderrEnabled       bool
+	StderrLevel         slog.Level
+	ElasticsearchConfig *ElasticsearchConfig
 }
 
 // LogfileConfig defines the configuration for file logging
 type LogfileConfig struct {
-	Dir    string
-	Prefix string
-	Level  slog.Level
+	Dir          string
+	Prefix       string
+	Level        slog.Level
+	Rotate       bool
+	RotatePeriod time.Duration
 }
 
 // ElasticsearchConfig defines the configuration for Elasticsearch logging
 type ElasticsearchConfig struct {
-	Addresses   []string
-	Username    string
-	Password    string
-	IndexPrefix string
-	Level       slog.Level
+	Addresses    string
+	Username     string
+	Password     string
+	IndexPrefix  string
+	Level        slog.Level
+	Rotate       bool
+	RotatePeriod time.Duration
 }
 
-// defaultConfig returns the default configuration
-func defaultConfig() *Config {
+// makeConfig returns the default configuration
+func makeConfig() *Config {
+	fileRotatePeriod, err := time.ParseDuration(config.Get().LogFileRotation)
```

Comment:
I get a panic at this line when calling Zeno (get url..) with `-h`:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x298 pc=0xdf38e1]

goroutine 1 [running]:
github.com/internetarchive/Zeno/internal/pkg/log.makeConfig()
	/home/corentin/Documents/work/ia/Zeno/internal/pkg/log/config.go:45 +0x21
github.com/internetarchive/Zeno/internal/pkg/log.Start.func1()
	/home/corentin/Documents/work/ia/Zeno/internal/pkg/log/log.go:26 +0x1b
sync.(*Once).doSlow(0x116904c?, 0x412985?)
	/home/corentin/.go/src/sync/once.go:76 +0xb4
sync.(*Once).Do(...)
	/home/corentin/.go/src/sync/once.go:67
github.com/internetarchive/Zeno/internal/pkg/log.Start()
	/home/corentin/Documents/work/ia/Zeno/internal/pkg/log/log.go:25 +0x45
main.main()
	/home/corentin/Documents/work/ia/Zeno/main.go:40 +0x3f
```

Reply:
Oh that's cause we run the pipeline from main and not from a cmd func (which is a bad practice)
So when you pass `-h`, `spf13/cobra`'s helper func runs and then the rest of the main is executed _BUT_ the config haven't been initialized and the call to `config.Get().<whatever>` segfaults cause the `config == nil`
We need to init the pipeline at the `cmd` level.

Reply:
Resolved by #170 and last commit